pipeline_alerts_email/handler.py

```diff
@@ -0,0 +1,86 @@
+import json
+import logging
+
+from okdata.sdk.data.dataset import Dataset
+
+from pipeline_alerts_email.mail import send_email
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def has_failed(message):
+    detail = message.get("detail", {})
+    return detail.get("status") in ["ABORTED", "FAILED", "TIMED_OUT"]
+
+
+def pipeline_input(message):
+    detail = message.get("detail", {})
+    return json.loads(detail.get("input", "{}"))
+
+
+def output_dataset(message):
+    return pipeline_input(message).get("output_dataset", {}).get("id")
+
+
+def dataset_contact_address(dataset):
+    ds = Dataset().get_dataset(dataset) or {}
+    return ds.get("contactPoint", {}).get("email")
+
+
+def handle_message(message):
+    logger.info(f"Handling message: {message}")
+
+    detail_type = message.get("detail-type")
+    if detail_type != "Step Functions Execution Status Change":
+        logger.info(f"Unexpected message detail-type: {detail_type}, skipping")
+        return
+
+    source = message.get("source")
+    if source != "aws.states":
+        logger.info(f"Unexpected message source: {source}, skipping")
+        return
+
+    if not has_failed(message):
+        logger.info("Pipeline hasn't failed, skipping")
+        return
+
+    dataset = output_dataset(message)
+    if not dataset:
+        logger.info("Missing output dataset, skipping")
+        return
+
+    contact_address = dataset_contact_address(dataset)
+    if not contact_address:
+        logger.info("Missing dataset contact address, skipping")
```

Comment:
Burde sende disse til dataplattform@oslo.kommune.no?

Reply:
Jeg tenkte at vi allerede får varslene til Slack?

Reply:
Ja, med mindre vi ønsker å slutte med det når dataeierne får det direkte :-) Men, vi kan evt. fikse det seinere!

Reply:
Ja, det var jo et poeng! Men ja, jeg tenker at vi kan beholde Slack-varslene litt til, siden vi ikke har beskrivende statusmeldinger i alle tilfeller enda, så da vil de nok fortsatt kontakte oss for å høre hva som gikk galt. :sweat_smile: